modules/templates/eval/eval.go

```diff
@@ -0,0 +1,343 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package eval
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"code.gitea.io/gitea/modules/util"
+)
+
+type Num struct {
+	Value any // int64 or float64, nil on error
+}
+
+var opPrecedence = map[string]int{
+	// "(": 1, this is for low precedence like function calls, they are handled separately
+	"or":  2,
+	"and": 3,
+	"not": 4,
+	"==":  5, "!=": 5, "<": 5, "<=": 5, ">": 5, ">=": 5,
+	"+": 6, "-": 6,
+	"*": 7, "/": 7,
```

Comment:
Just a question.

While `equal` is `==`  instead of `eq`, why is `or` represented by `or` instead of `||`? And `and`/`&&`, `not`/`!`.

So it's more python style than go style, right? 

BTW, I think we need a document to list supported operators and functions, or I have to read the code to understand how to use `Eval`.

Reply:
`&&` and `||` in most languages:  `a() && b()` if `a()` is false, then `b()` is not called.

So we choose `and` / `or`, to show that `a() and b()` will call `a()` and `b()` both. Then `not` also matches these names.

More comments in be88baf020286344c5ec4b7a036de8a837cdfced

Reply:
Although we can add support for `ne`, `eq`, `lt`, `gt` as aliases for `!=`, `==`, `<`, `>`, I think it's not necessary.

These `ne`, `eq`, `lt`, `gt` words  are just Golang's bypass design  to support some operations, since we can have clearer `!=`, `==`, `<`, `>`, we do not need these aliases.